external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -193,7 +197,56 @@ pub(crate) fn write_prologue(
     })?;
 
     write!(file, "{}", prologue)?;
+    Ok(())
+}
+
+pub fn update_dependency_graph(
+    file: &mut LockFile,
+    manifest_digest: String,
+    deps_digest: String,
+    dependencies: Option<toml_edit::Value>,
+    dev_dependencies: Option<toml_edit::Value>,
+    packages: Option<ArrayOfTables>,
+) -> Result<()> {
```

Comment:
Updates the relevant TOML values for the dependency graph in `file` without affecting any other part of the contents.

Currently, `file` passed here is always generated from scratch (as before). But in upcoming changes, `file` will be an _existing_ lock file, e.g., containing automated address management info, that will now remain intact when the dependency graph is updated via this function.